From 4cc7954a49958cf8e6ebdc3de85f6070a24e910c Mon Sep 17 00:00:00 2001
From: Oleh Nykyforchyn <oleh.nyk@gmail.com>
Date: Thu, 7 Aug 2025 18:39:38 +0300
Subject: [PATCH] xserver: os: reexport TimerForce() but do not allow compile
 drivers with it

This function is called by nvidia390 legacy proprietary driver.

Signed-off-by: Oleh Nykyforchyn <oleh.nyk@gmail.com>
---
 os/osdep.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/os/osdep.h b/os/osdep.h
index 76202226b..7dda33f19 100644
--- a/os/osdep.h
+++ b/os/osdep.h
@@ -118,7 +118,12 @@ extern Bool ComputeLocalClient(ClientPtr client);
 
 /* OsTimer functions */
 void TimerInit(void);
-Bool TimerForce(OsTimerPtr timer);
+
+/* must be exported for backwards compatibility with legacy nvidia390,
+ * not for use in maintained drivers
+ */
+
+extern _X_EXPORT Bool TimerForce(OsTimerPtr);
 
 #ifdef WIN32
 #include <X11/Xwinsock.h>
-- 
2.47.1

